tests/rkt_auth_test.go

```diff
@@ -0,0 +1,289 @@
+	runRktBang(t, server.URL, "basic-custom-config")
+}
+
+func basicWithVendorConfig(t *testing.T, server *taas.Server) {
+	removeAllConfig(t)
+	writeVendorConfig(t, "test.json", server.Conf)
+	runRktBang(t, server.URL, "basic-vendor-config")
+}
+
+func TestAuthOauth(t *testing.T) {
+	skipUnsafe(t)
+	removeDataDir(t)
+	defer removeAllConfig(t)
+	server := runServer(t, taas.Oauth)
+	defer server.Close()
+	oauthNoConfig(t, server)
+	oauthCustomConfig(t, server)
+	oauthVendorConfig(t, server)
+}
+
+func oauthNoConfig(t *testing.T, server *taas.Server) {
+	removeAllConfig(t)
+	runRkt401(t, server.URL, "oauth1")
+}
+
+func oauthCustomConfig(t *testing.T, server *taas.Server) {
+	removeAllConfig(t)
+	writeCustomConfig(t, "test.json", server.Conf)
+	runRktBang(t, server.URL, "oauth-custom-config")
+}
+
+func oauthVendorConfig(t *testing.T, server *taas.Server) {
+	removeAllConfig(t)
+	writeVendorConfig(t, "test.json", server.Conf)
+	runRktBang(t, server.URL, "oauth-vendor-config")
+}
+
+func TestAuthOverride(t *testing.T) {
+	skipUnsafe(t)
+	removeDataDir(t)
+	defer removeAllConfig(t)
+	server := runServer(t, taas.Oauth)
+	defer server.Close()
+	validVendorInvalidCustom(t, server)
+	invalidVendorValidCustom(t, server)
+}
+
+func validVendorInvalidCustom(t *testing.T, server *taas.Server) {
+	removeAllConfig(t)
+	writeVendorConfig(t, "test.json", server.Conf)
+	runRktBang(t, server.URL, "oauth-vvic-1")
+	writeCustomConfig(t, "test.json", getInvalidOAuthConfig(server.Conf))
+	runRkt401(t, server.URL, "oauth-vvic-2")
+}
+
+func invalidVendorValidCustom(t *testing.T, server *taas.Server) {
+	removeAllConfig(t)
+	writeVendorConfig(t, "test.json", getInvalidOAuthConfig(server.Conf))
+	runRkt401(t, server.URL, "oauth-ivvc-1")
+	writeCustomConfig(t, "test.json", server.Conf)
+	runRktBang(t, server.URL, "oauth-ivvc-2")
+}
+
+func TestAuthIgnore(t *testing.T) {
+	skipUnsafe(t)
+	removeDataDir(t)
+	defer removeAllConfig(t)
+	server := runServer(t, taas.Oauth)
+	defer server.Close()
+	bogusFiles(t, server)
+	subdirectories(t, server)
+}
+
+func bogusFiles(t *testing.T, server *taas.Server) {
+	removeAllConfig(t)
+	writeVendorConfig(t, "README", "This is vendor config")
+	writeCustomConfig(t, "README", "This is custom config")
+	writeVendorConfig(t, "test.notjson", server.Conf)
+	writeCustomConfig(t, "test.notjson", server.Conf)
+	runRkt401(t, server.URL, "oauth-bogus-files")
+}
+
+func subdirectories(t *testing.T, server *taas.Server) {
+	removeAllConfig(t)
+	customSubdir := filepath.Join(config.DefaultCustomPath, "subdir")
+	vendorSubdir := filepath.Join(config.DefaultVendorPath, "subdir")
+	writeConfig(t, customSubdir, "test.json", server.Conf)
+	writeConfig(t, vendorSubdir, "test.json", server.Conf)
+	runRkt401(t, server.URL, "oauth-subdirectories")
+}
+
+// TODO (krnowak): Remove this when we will be able to specify
+// different custom and vendor configuration directories.
+const unsafeEnvVar = "DELETE_DATA_AND_CONFIG"
```

Comment:
I suggest `const enableDestructiveTestsEnvVar = RKT_ENABLE_DESTRUCTIVE_TESTS`

`skipUnsafe` and `removeDataDir` are always used together (for a good reason). Maybe the tests could only call `removeDataDir` but the body of `skipUnsafe` could be moved in `removeDataDir`?